griptape/tasks/prompt_task.py

```diff
@@ -98,12 +104,33 @@ def after_run(self) -> None:
             self.id,
             self.output.to_text() if self.output is not None else "",
         )
+        structure = self.structure
+        if (
+            structure is not None
+            and structure.conversation_memory_strategy == structure.ConversationMemoryStrategy.PER_TASK
+            and self.conversation_memory is not None
+            and self.output is not None
+        ):
+            run = Run(input=self.input, output=self.output)
+
+            self.conversation_memory.add_run(run)
 
     def try_run(self) -> BaseArtifact:
         message = self.prompt_driver.run(self.prompt_stack)
 
         return message.to_artifact()
 
+    def preprocess(self, structure: Structure) -> BaseTask:
+        super().preprocess(structure)
+
+        if self.conversation_memory is NOTHING:
+            if structure.conversation_memory is not None:
+                self.conversation_memory = structure.conversation_memory
```

Comment:
double checking, the `NOTHING` addition is just for this check here?

Reply:
Yes. Since we need to support the user not setting conversation memory (`NOTHING`) _and_ the user explicitly disabling conversation memory (`None`).